torch/csrc/jit/serialization/export_bytecode.cpp

```diff
@@ -1,23 +1,333 @@
+    Method cur = stack.front();
+    stack.pop_front();
+    auto tup = std::make_pair(
+        cur.owner()._ivalue()->type()->name()->qualifiedName(),
+        &cur.function());
+    if (visited.find(tup) != visited.end()) {
+      continue;
+    }
+    visited.insert(tup);
+    const auto& f = toGraphFunction(cur.function());
+    auto graph = f.graph()->copyUnique();
+    Inline(*graph);
+    c10::QualifiedName qn(*cur.owner()._ivalue()->type()->name(), f.name());
+
+    if (incl_dependent_functions) {
+      std::vector<Method> dependent_methods =
+          findAllDependentFunctions(cur.owner(), *graph);
+      std::copy(
+          dependent_methods.begin(),
+          dependent_methods.end(),
+          std::back_inserter(stack));
+    }
+    auto inlined_func = std::make_unique<GraphFunction>(
+        qn, std::move(graph), f.function_creator());
+    inlined_func->setSchema(f.getSchema());
+    inlined_functions.emplace_back(std::move(inlined_func));
+  }
+  return inlined_functions;
+}
+
+std::unique_ptr<mobile::Code> compileGraphToMobileCode(
+    const std::string& name,
+    const std::shared_ptr<Graph>& graph,
+    const CompilationOptions& compilation_options,
+    BackendDebugInfoRecorder& debug_info_recorder) {
+  MobileCode code(
+      graph,
+      name,
+      compilation_options.enable_default_value_for_unspecified_arg,
+      compilation_options.enable_default_args_before_out_args);
+
+  std::unique_ptr<mobile::Code> mobile_code_ptr =
+      std::make_unique<mobile::Code>();
+  mobile::Code& mobile_code = *mobile_code_ptr;
+
+  // operator names
+  std::vector<std::string> method_names;
+  std::vector<int64_t> op_debug_handles;
+  int next_new_op_index = 0;
+
+  auto op_to_specified_args = code.op_to_num_specified_args();
+
+  for (size_t i = 0; i < code.instructions().size(); ++i) {
+    Instruction ins = code.instructions()[i];
+
+    if ((ins.op == OP || ins.op == OPN) && ins.X == next_new_op_index) {
+      // Found a new op (assumes new operators ordered by ascending ins.X)
+      auto node = code.instructions_source()[i];
+      const c10::OperatorName& opname = node->schema().operator_name();
+      auto unique_name = c10::toString(opname);
+      // For operator with vararg, adding default arguments would be confusing
+      // and is not allowed. For an operator with num_args = -1, it means the
+      // number of arguments is not available for this operator, we don't do any
+      // backward compatibility adaptation at runtime.
+      c10::optional<int> num_args = c10::nullopt;
+      auto it = op_to_specified_args.find(unique_name);
+      if (it != op_to_specified_args.end()) {
+        num_args = it->second;
+      }
+      mobile_code.operator_input_sizes_.emplace_back(num_args.value_or(-1));
+      mobile_code.op_names_.emplace_back(opname);
+      auto func = mobile::makeOperatorFunction(
+          opname, num_args, compilation_options.model_version);
+      TORCH_INTERNAL_ASSERT(
```

Comment:
We are not exporting yet. 

This is converting Graph -> bytecode in memory. So the operator has to exist. The operator exists because it is bound in the Graph's Node already. Alternatively we could use the Operator* inside of Node instead of doing an lookup with name; but then we still need something to wrap it to std::function so the extra saving of one less lookup is not worth it.